File: 백준/Bronze/11050. 이항 계수 1/이항 계수 1.py
import sys

input = sys.stdin.readline
write = sys.stdout.write

def solve():
    line = input().split()
    if not line: return
    
    n, k = map(int, line)
    if not 1 <= n <= 10: return
    if not 0 <= k <= n: return
    
    nfac = 1
    for i in range(1, n + 1):
        nfac *= i
    
    kfac = 1
    for i in range(1, k + 1):
        kfac *= i
        
    nkfac = 1
    for i in range(1, n - k + 1):
        nkfac *= i
        
    result = nfac // (kfac * nkfac)
    
    write(str(result) + "\n")
    
if __name__ == "__main__":
    solve()

File: 백준/Bronze/11050. 이항 계수 1/이항 계수 1.cc
#include <stdio.h>

int fac(int num) {
	int val = 1;
	for (int i = 1; i <= num; i++) {
		val *= i;
	}

	return val;
}

int main(void) {
	int n = 0, k = 0;
	if (scanf("%d %d", &n, &k) != 2) return 0;
	if (1 > n || n > 10 || 0 > k || k > n) return 0;

	int nfac = fac(n);
	int kfac = fac(k);
	int nkfac = fac(n - k);

	int result = nfac / (kfac * nkfac);

	printf("%d\n", result);

	return 0;
}